monkey/common/di_container.py

```diff
@@ -56,12 +61,45 @@ def register_instance(self, interface: Type[T], instance: T):
         self._instance_registry[interface] = instance
         DIContainer._del_key(self._type_registry, interface)
 
+    def register_convention(self, type_: Type[T], name: str, instance: T):
+        """
+        Register an instance as a convention
+
+        At times — particularly when dealing with primative types — it can be useful to define a
+        convention for how dependencies should be resolved. For example, you might want any class
+        that specifies `hostname: str` in its constructor to receive the hostname of the system it's
+        running on. Registering a convention allows you to assign an object instance to a type, name
+        pair.
```

Comment:
```suggestion
        running on. Registering a convention allows you to assign an object instance to a tuple of type and name.
```

Reply:
`tuple` implies something about the implementation, which the documentation should avoid. "pair" is more generic.